.github/workflows/document.yml

```diff
@@ -0,0 +1,275 @@
+          'restify',
+          'tedious',
+          'vision',
+          'winston',
+          # core modules
+          'crypto',
+          'fs',
+          'http',
+          'https',
+          'zlib'
+        ]
+
+    steps:
+      # the working directory is created by the runner and mounted to the container.
+      # container user is root and the runner is not a user in the container.
+      # this is a github actions design flaw.
+      # when npm 7 is run as root, scripts are always run with the effective uid and gid of the working directory owner.
+      # node 16 can't install under default setup.
+      # specifying workdir for container and path for checkout does not work due to bug.
+      - name: Change Owner of Container Working Directory
+        run: chown root:root .
+
+      - name: Checkout ${{ github.ref }}
+        uses: actions/checkout@v2
+
+      - name: Setup Node ${{ matrix.node }}
+        uses: actions/setup-node@v2
+        with:
+          node-version: ${{ matrix.node  }}
+
+      - name: Show Environment Info
+        run: |
+          printenv
+          node --version
+          npm --version 
+          cat /etc/os-release
+
+      - name: NPM Install
+        run: npm install  --unsafe-perm
+
+      - name: Create tmp Directory
+        run: |
+          mkdir -p docs/tmp
+
+      - name: Test ${{ matrix.pkg }} versions
+        # in testeachversion we may fail tests, and want to keep going.
+        continue-on-error: true
+        # some packages take a long time.
+        timeout-minutes: 30
+        run: npx testeachversion --verbose -l ./docs/tmp -p ${{ matrix.pkg }}
+
+      - name: Rename Result Artifacts
+        # substitution needed because:
+        # 1. files contain characters that are illegal for uploaded files
+        # 2. we need to keep a specific file name pattern for testeachversion to process later
+        # 3. we want to use matrix data to tag the files
+        # bash needed for substitution. 
+        shell: bash
+        run: .github/scripts/testeachversion-rename-files.sh docs/tmp ${{ matrix.node }} ${{ matrix.pkg }}
+
+      - name: Upload Artifacts
+        uses: actions/upload-artifact@v2
+        with:
+          name: supported-components-data
+          path: docs/tmp/ # or path/to/artifact
+
+
+  create-pull-request:
+    name: Create Pull Request
+    runs-on: ubuntu-latest # environment job will run in
+    needs:  test-each-version
+
+    steps:
+      - name: Checkout ${{ github.ref }}
+        uses: actions/checkout@v2
+
+      - name: Init Git
+        run: |
+         git config user.name "GitHub Actions"
+         git config user.email noreply@github.com
+
+      # job is running on the github runner not in pre-configured container
+      - name: Set Time Zone & Current Date
+        id: date
+        run: |
+          sudo timedatectl set-timezone America/Los_Angeles
+          echo "::set-output name=date::$(date +'%Y-%m-%d-%H-%M-%S')" # must be valid branch name
```

Comment:
Minor, is this echo needed?

Reply:
Yep. That's how it is done. See here: https://docs.github.com/en/actions/reference/workflow-commands-for-github-actions.

Reply:
Ah yes, closer read and now I see you're setting the value of `steps.date.outputs.date`... I mistook this step as just setting a correct timezone on the runner :)